pulp_ansible/app/models.py

```diff
@@ -332,15 +332,9 @@ class AnsibleNamespaceMetadata(Content):
 
     @property
     def avatar_artifact(self):
-        if self.avatar_sha256:
-            avatar = self._artifacts.filter(sha256=self.avatar_sha256).first()
-            if avatar is None:
-                log.debug(
-                    f"Artifact({self.avatar_sha256}) is missing for namespace avatar "
-                    f"{self.name}:{self.metadata_sha256}"
-                )
-            return avatar
-
+        ca = self.contentartifact_set.filter(relative_path=f"{self.name}-avatar").first()
```

Comment:
Can you check if we need a `select_related('artifact')`  here?

Reply:
I updated the query to be about artifacts without indirection.